kraken/network_chaos/actions.py

```diff
@@ -0,0 +1,213 @@
+            joblst = []
+            egress_lst = [i for i in param_lst if i in test_egress]
+            chaos_config = {
+                "network_chaos": {
+                    "duration": test_duration,
+                    "interfaces": test_interface,
+                    "node_name": ",".join(nodelst),
+                    "execution": test_execution,
+                    "instance_count": test_instance_count,
+                    "egress": test_egress,
+                }
+            }
+            logging.info("Executing network chaos with config \n %s" % yaml.dump(chaos_config))
+            try:
+                for i in egress_lst:
+                    for node in nodelst:
+                        exec_cmd = get_egress_cmd(
+                            test_execution, test_interface, i, test_dict["egress"], duration=test_duration
+                        )
+                        logging.info("Executing %s on node %s" % (exec_cmd, node))
+                        job_body = yaml.safe_load(
+                            job_yml.format(jobname=i + str(hash(node))[:5], nodename=node, cmd=exec_cmd)
+                        )
+                        joblst.append(job_body["metadata"]["name"])
+                        api_response = kubecli.create_job(job_body)
+                        if api_response is None:
+                            raise Exception("Error creating job")
+                    if test_execution == "serial":
+                        logging.info("Waiting for serial job to finish")
+                        start_time = int(time.time())
+                        wait_for_job(joblst[:], test_duration + 300)
+                        logging.info("Waiting for wait_duration %s" % wait_duration)
+                        time.sleep(wait_duration)
+                        end_time = int(time.time())
+                        cerberus.publish_kraken_status(config, failed_post_scenarios, start_time, end_time)
+                    if test_execution == "parallel":
+                        break
+                if test_execution == "parallel":
+                    logging.info("Waiting for parallel job to finish")
+                    start_time = int(time.time())
+                    wait_for_job(joblst[:], test_duration + 300)
+                    logging.info("Waiting for wait_duration %s" % wait_duration)
+                    time.sleep(wait_duration)
+                    end_time = int(time.time())
+                    cerberus.publish_kraken_status(config, failed_post_scenarios, start_time, end_time)
+            except Exception as e:
+                logging.error("Network Chaos exiting due to Exception %s" % e)
+                sys.exit(1)
+            finally:
+                logging.info("Deleting jobs")
+                delete_job(joblst[:])
+
+
+def verify_interface(test_interface, nodelst):
+    pod_body = yaml.safe_load(pod_yml.format(nodename=nodelst[0]))
```

Comment:
can we randomly select a node here to use? Maybe also set it to a variable to usee in the later steps (line 153)

Reply:
sure will do, will pick a node at random.